ch.sbb.scion.rcp.microfrontend.e3.app.demo/src/ch/sbb/scion/rcp/microfrontend/e3/app/demo/popup/TestPopupDialog.java

```diff
@@ -0,0 +1,124 @@
+package ch.sbb.scion.rcp.microfrontend.e3.app.demo.popup;
+
+import org.eclipse.jface.dialogs.Dialog;
+import org.eclipse.jface.dialogs.IDialogConstants;
+import org.eclipse.jface.layout.GridDataFactory;
+import org.eclipse.jface.widgets.LabelFactory;
+import org.eclipse.jface.widgets.TextFactory;
+import org.eclipse.swt.SWT;
+import org.eclipse.swt.events.SelectionAdapter;
+import org.eclipse.swt.events.SelectionEvent;
+import org.eclipse.swt.graphics.Point;
+import org.eclipse.swt.widgets.Composite;
+import org.eclipse.swt.widgets.Control;
+import org.eclipse.swt.widgets.Shell;
+
+import com.google.gson.GsonBuilder;
+
+import ch.sbb.scion.rcp.workbench.ISciWorkbenchPopup;
+
+public class TestPopupDialog extends Dialog {
+
+  private static final int CLOSE_WITH_EXCEPTION_ID = IDialogConstants.CLIENT_ID + 1;
+
+  private final ISciWorkbenchPopup popup;
+
+  private boolean activated = false;
+
+  protected TestPopupDialog(final Shell parentShell, final ISciWorkbenchPopup popup) {
+    super(parentShell);
+    this.popup = popup;
+
+    configureShellStyle();
+  }
+
+  @Override
+  protected Control createDialogArea(final Composite parent) {
+    var area = (Composite) super.createDialogArea(parent);
+    var gson = new GsonBuilder().setPrettyPrinting().create();
+
+    LabelFactory.newLabel(SWT.NONE).text("Capability:").create(area);
+    var capabilityText = TextFactory.newText(SWT.BORDER | SWT.WRAP | SWT.READ_ONLY | SWT.V_SCROLL).create(area);
+    GridDataFactory.fillDefaults().grab(true, true).applyTo(capabilityText);
+    capabilityText.setText(gson.toJson(popup.getCapability()));
+
+    LabelFactory.newLabel(SWT.NONE).text("Parameters:").create(area);
+    var paramsText = TextFactory.newText(SWT.BORDER | SWT.WRAP | SWT.READ_ONLY | SWT.V_SCROLL).create(area);
+    GridDataFactory.fillDefaults().grab(true, false).hint(new Point(SWT.DEFAULT, 100)).applyTo(paramsText);
+    paramsText.setText(gson.toJson(popup.getParams()));
+
+    LabelFactory.newLabel(SWT.NONE).text("Referrer View ID:").create(area);
+    var referrerText = TextFactory.newText(SWT.BORDER | SWT.READ_ONLY).create(area);
+    GridDataFactory.fillDefaults().grab(true, false).applyTo(referrerText);
+    referrerText.setText(popup.getReferrerViewId().orElse("-"));
+    return area;
+  }
+
+  @Override
+  protected void okPressed() {
+    super.okPressed();
```

Comment:
do we really want to let swt.Dialog handle okPressed first before we "close" the popup? I am always not sure what to do in these destructive methods